src/pke/lib/encoding/ckkspackedencoding.cpp

```diff
@@ -374,6 +374,160 @@ bool CKKSPackedEncoding::Encode() {
 }
 #endif
 
+bool CKKSPackedEncoding::EncodeWithExtra() {
+  if (this->isEncoded) return true;
+
+  uint32_t N = GetElementRingDimension();
+  uint32_t Nh = (N >> 1);
+  std::vector<std::complex<double>> inverse = value;
+  inverse.resize(Nh);
+  DiscreteFourierTransform::FFTSpecialInv(inverse);
+
+  int64_t q = 9223372036854775295;
+
+  if (this->typeFlag == IsDCRTPoly) {
+    const std::shared_ptr<ILDCRTParams<BigInteger>> params =
+        this->encodedVectorDCRT.GetParams();
+    const std::vector<std::shared_ptr<ILNativeParams>> &nativeParams =
+        params->GetParams();
+    usint numTowers = nativeParams.size();
+    std::vector<DCRTPoly::Integer> moduli(numTowers);
+    for (usint i = 0; i < numTowers; i++) {
+      moduli[i] = nativeParams[i]->GetModulus();
+    }
+
+    int32_t logc = 0, logci;
+    for (size_t i = 0; i < Nh; ++i) {
+      inverse[i] *= scalingFactor;
+      logci = static_cast<int32_t>(ceil(log2(abs(inverse[i].real()))));
```

Comment:
logci should be defined here instead of line 399

Reply:
fixed